File: web/page2reader/war/closure/wit/page2reader/model/pageurlobj.js
goog.provide('wit.page2reader.model.PageUrlObj');

goog.require('wit.base.constants');
goog.require('wit.page2reader.constants');



/**
 * PageUrl object.
 * @constructor
 */
wit.page2reader.model.PageUrlObj = function() {
};


/** @type {string} */
wit.page2reader.model.PageUrlObj.prototype.keyString;


/** @type {string} */
wit.page2reader.model.PageUrlObj.prototype.pUrl;


/** @type {string} */
wit.page2reader.model.PageUrlObj.prototype.serverCreateTimeMillis;


/** @type {string} */
wit.page2reader.model.PageUrlObj.prototype.title;


/** @type {string} */
wit.page2reader.model.PageUrlObj.prototype.text;


/** @type {string} */
wit.page2reader.model.PageUrlObj.prototype.cleansedPage;


/** @type {number} */
wit.page2reader.model.PageUrlObj.prototype.sentCount;


/**
 * Set contents from JSON string.
 * @param {!string} jsonString JSON string.
 */
wit.page2reader.model.PageUrlObj.prototype.setContentsFromJSONString = function(
    jsonString) {

  var jsonObj = goog.json.unsafeParse(jsonString);
  this.keyString = jsonObj[wit.page2reader.constants.KEY_STRING];
  this.pUrl = jsonObj[wit.page2reader.constants.P_URL];
  this.serverCreateTimeMillis = jsonObj[
      wit.page2reader.constants.SERVER_CREATE_TIME_MILLIS];
  this.title = jsonObj[wit.page2reader.constants.TITLE];
  this.text = jsonObj[wit.page2reader.constants.TEXT];
  this.cleansedPage = jsonObj[wit.page2reader.constants.CLEANSED_PAGE];
  this.sentCount = jsonObj[wit.page2reader.constants.SENT_COUNT];
};

File: android/src/com/wit/page2reader/model/PageUrlObj.java
package com.wit.page2reader.model;

import java.util.ArrayList;

import org.json.JSONObject;

import com.wit.page2reader.Constants;
import com.wit.page2reader.Constants.DelBtnStatus;
import com.wit.page2reader.Constants.ResendBtnStatus;

public class PageUrlObj {

    public String keyString;
    public String pUrl;
    public long serverCreateTimeMillis;
    public String title;
    public String text;
    public int sentCount;

    public DelBtnStatus delBtnStatus = DelBtnStatus.NORMAL;
    public ResendBtnStatus resendBtnStatus = ResendBtnStatus.NORMAL;

    public PageUrlObj(JSONObject jsonObject) {
        this.keyString = jsonObject.getString(Constants.KEY_STRING);
        this.pUrl = jsonObject.getString(Constants.P_URL);
        this.serverCreateTimeMillis = jsonObject.getLong(Constants.SERVER_CREATE_TIME_MILLIS);
        this.title = jsonObject.getString(Constants.TITLE);
        this.text = jsonObject.getString(Constants.TEXT);
        this.sentCount = jsonObject.getInt(Constants.SENT_COUNT);
    }

    public static PageUrlObj getPageUrl(ArrayList<PageUrlObj> pageUrls, String keyString) {
        for (PageUrlObj pageUrl : pageUrls) {
            if (pageUrl.keyString.equals(keyString)) {
                return pageUrl;
            }
        }
        return null;
    }
}
